### DISCUSSION TITLE
Questions about compression

### DISCUSSION THREAD

**Discussion by AlexZIX** [minio/minio#13943] [Q&A] [answered]
After reading docs about how compression works in minio I still have 2 unanswered questions:
1. We can set list of files extensions and mime-types for compression. Should uploaded file met both criteria or may be just one of them to be compressed?
2. Where I can see if file stored compressed or not? Also it will be helpful to see the information about compression ratio.

**Comment by klauspost** [comment, marked as answer]
1. If either extension or mime type matches, files are compressed.
2. That information is only available internally.

**Reply by AlexZIX** [reply]
Think it's may helpful to get this information using mc.

**Reply by klauspost** [reply]
@AlexZIX Exposing it via standard API calls is not something we are looking to do since compression can optionally be combined with encryption and we don't want to expose compressed sizes.

We could add it as part of bucket statistics, but it is not a top priority.

**Reply by AlexZIX** [reply]
I understand that it's not main priority but will be usefull even in buckert statistics.

**Reply by StormYudi** [reply]
I agree, we need this in bucket statistics.
